library/src/main/java/com/google/android/exoplayer/text/SubtitleParserHelper.java

```diff
@@ -179,8 +179,13 @@ private void handleSample(long sampleTimeUs, SampleHolder holder) {
       if (sampleHolder != holder) {
         // A flush has occurred since this holder was posted. Do nothing.
       } else {
-        this.result = new PlayableSubtitle(parsedSubtitle, subtitlesAreRelative, sampleTimeUs,
-            subtitleOffsetUs);
+        // it's more accurate to read startTime from eventTime compare to sampleTime
```

Comment:
Actually, you still need startTimeUs for the absolute timestamp case. But you could do the following here:

```
        this.result = parsedSubtitle.getEventTimeCount() == 0 ? null : new PlayableSubtitle(
            parsedSubtitle, subtitlesAreRelative, sampleTimeUs, subtitleOffsetUs);
```

And then implement the PlayableSubtitle constructor as:

```
    this.subtitle = subtitle;
    this.offsetUs = (isRelative ? startTimeUs : 0) + offsetUs;
    this.firstEventTimeUs = getEventTime(0);
```

Where `firstEventTimeUs` replaces `startTimeUs` as the member variable.